docs/uri-formattering.md

```diff
@@ -0,0 +1,61 @@
+
+![client API communicatie zonder uri-format negotiation](img/1-communicatie-client-api-zonder-uri-negotiation.svg)
+
+De Haal Centraal APIs maken gebruik van [URI Template](https://datatracker.ietf.org/doc/html/rfc6570) om urls te parameteriseren. Hiermee kunnen gelijksoortige urls (urls die alleen verschillen door path parameter waarde) worden gerepresenteerd met één URI template.
+
+Een URI template bevat één of meerdere expressions waarmee een verzameling van gelijksoortige URIs kan worden beschreven. Een expression is de tekst tussen een open accolade '{' en een sluit accolade '}' inclusief de accolade tekens. Er zijn meerdere expression types met bijbehorende expansie methode. Zie [Expressions](https://datatracker.ietf.org/doc/html/rfc6570#section-2.2).
+De Haal Centraal APIs gebruiken 'no operator expressions' zodat consumers 'Simple String Expansion'
+kunnen toepassen. Dit houdt in dat een {expression} zonder toepassing van logica kan worden vervangen met een waarde.
+Binnen de Haal Centraal APIs worden property namen gebruikt voor expressions. Een property naam is:
+- de naam van een property van een resource, v.b. burgerservicenummer of
+- de naam van een property van een gegevensgroep van een resource v.b. kinderen.burgerservicenummer
+
+Om de logica voor het aanpassen van Haal Centraal urls door Backend for Frontends en API Gateways zo simpel mogelijk te houden, en om de Haal Centraal API clients de mogelijkheid te bieden om hun gewenste uri formaat aan te kunnen geven, kunnen de Haal Centraal APIs gebruik maken van
+
+- [server-driven negotiation](https://developer.mozilla.org/en-US/docs/Web/HTTP/Content_negotiation)
+- uri formats voor [Resource Identifiers](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.5)
+- [uri-template](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.7.3.6) 
+
+om de urls in een response te formatteren.
+
+Door middel van een custom HTTP header 'Accept-Uri-Format' kan een Haal Centraal API consumer in een request aangeven in welke format hij de urls in de response wil ontvangen.
+
+De volgende uri formats worden ondersteund:
+- uri. Alle links in de response worden geformatteerd als absolute urls.
+- uri-reference. 'Same API' links worden geformatteerd als relatieve urls en 'externe API' links als absolute urls.
+- uri-template. De server deel van urls wordt gerepresenteerd door een URI Template [Expression](https://datatracker.ietf.org/doc/html/rfc6570#section-2.2).
+  Als convententie wordt gehanteerd dat de url van het Server object in de OpenApi specificatie van een Haal Centraal API wordt gerepresenteerd door één van de de volgende server url expressions:
+  - {brpserverurl}
+  - {brphistorieserverurl}
+  - {brpbewoningserverurl}
+  - {bagserverurl}
```

Comment:
Ik mis BRP reisducumenten in dit lijstje

Reply:
@MelvLee kan je in feature ook voorbeeld opnemen dat wel base url wordt beëinvloed, maar niet een identificatie-template? Bijvoorbeeld ook met Uri kan je krijgen:
_links: kinderen: [ href: "https://www.gemeente.nl/api/brp/personen/{uuid}"]